spring-pulsar/src/test/java/org/springframework/pulsar/core/CachingPulsarProducerFactoryTests.java

```diff
@@ -111,41 +113,68 @@ void createProducerWithMatrixOfCacheKeys() throws PulsarClientException {
 		Schema<String> schema2 = new StringSchema();
 		MessageRouter router1 = mock(MessageRouter.class);
 		MessageRouter router2 = mock(MessageRouter.class);
+		List<ProducerInterceptor> interceptors1 = List.of(mock(ProducerInterceptor.class));
+		List<ProducerInterceptor> interceptors2 = List.of(mock(ProducerInterceptor.class));
 
 		PulsarProducerFactory<String> producerFactory = producerFactory(pulsarClient, Collections.emptyMap());
 
-		// ask for the same 9 unique combos 3x - should end up w/ only 9 entries in cache
+		// ask for the same 21 unique combos 3x - should end up w/ only 21 entries in
```

Comment:
Ugghhh... permutations 🤣 

First it's 3. Then it's 9. Then it's 21. Then it's...

Reply:
Yeah I didn't like that part either but I wanted to cover the interceptor cases